src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    protected Figure getRandomFigureName() {
+        int maxSize = 20;
+        int minSize = 5;
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        int index = random.nextInt(Forms.values().length);
+        String value = Forms.values()[index].getForm();
+        switch (value) {
+            case "CIRCLE":
+                int radius = random.nextInt(maxSize) + minSize;
+                Figure circle = new Circle(radius);
+                circle.setColor(colorSupplier.getRandomColor());
+                circle.getArea();
+                circle.draw();
+                return circle;
+            case "RECTANGLE":
+                int widht = random.nextInt(maxSize) + minSize;
+                int height = random.nextInt(maxSize) + minSize;
+                Figure rectangle = new Rectangle(widht, height);
+                rectangle.setColor(colorSupplier.getRandomColor());
+                rectangle.getArea();
+                rectangle.draw();
+                return rectangle;
+            case "SQUARE":
+                int side = random.nextInt(maxSize) + minSize;
+                Figure square = new Square(side);
+                square.setColor(colorSupplier.getRandomColor());
+                square.getArea();
+                square.draw();
+                return square;
+            case "TRIANGLE":
+                int firstLeg = random.nextInt(maxSize) + minSize;
+                int secondLeg = random.nextInt(maxSize) + minSize;
+                Figure rightTriangle = new RightTriangle(firstLeg, secondLeg);
+                rightTriangle.setColor(colorSupplier.getRandomColor());
+                rightTriangle.getArea();
+                rightTriangle.draw();
+                return rightTriangle;
+            case "TRAPEZOID" :
+                int basisDown = random.nextInt(maxSize) + minSize;
+                int basisUp = random.nextInt(basisDown - 1) + basisDown - 2;
+                int height1 = random.nextInt(maxSize) + minSize;
+                Figure isoscelesTrapezoid = new IsoscelesTrapezoid(basisDown, basisUp, height1);
+                isoscelesTrapezoid.setColor(colorSupplier.getRandomColor());
+                isoscelesTrapezoid.getArea();
+                isoscelesTrapezoid.draw();
+                return isoscelesTrapezoid;
+            default:
+                return null;
+        }
+
+    }
+
+    protected Figure getDefaultFigure() {
+        int defaultSize = 10;
+        String defaultColor = "WHITE";

```

Comment:
not fixed, should be class constants (like this: private static final CONSTANT_CASE )